OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoLogic.cs

```diff
@@ -99,6 +99,28 @@ public GameInfoLogic(Widget widget, World world, IngameInfoPanel activePanel)
 					activePanel = IngameInfoPanel.Debug;
 			}
 
+			if (world.LobbyInfo.Clients.Count(c => c.Slot != null && c.Bot == null) > 1)
+			{
+				numTabs++;
+				var chatPanelContainer = widget.Get<ContainerWidget>("CHAT_PANEL");
+				var chatTabButton = widget.Get<ButtonWidget>(string.Concat("BUTTON", numTabs.ToString()));
+				chatTabButton.Text = "Chat";
+				chatTabButton.IsVisible = () => numTabs > 1 && !hasError;
+				chatTabButton.IsHighlighted = () => activePanel == IngameInfoPanel.Chat;
+				chatTabButton.OnClick = () =>
+				{
+					activePanel = IngameInfoPanel.Chat;
+					chatPanelContainer.Get<TextFieldWidget>("CHAT_TEXTFIELD").TakeKeyboardFocus();
+				};
+
+				chatPanelContainer.IsVisible = () => activePanel == IngameInfoPanel.Chat;
+
+				Game.LoadWidget(world, "CHAT_CONTAINER", chatPanelContainer, new WidgetArgs() { { "isMenuChat", true } });
+
+				if (activePanel == IngameInfoPanel.AutoSelect || world.IsGameOver)
+					chatTabButton.OnClick();
```

Comment:
A separate PR that could offer a good compromise here would be to remember and restore the last active tab when closing/reopening the menu (except for the game-end auto-open case, which should force the objectives tab).  This could be done by adding an `IngameInfoPanel` argument to the `onExit` handler so that `MenuButtonsChromeLogic` could save and reuse it in the `activePanel` argument.